### ISSUE TITLE
config templating not populating some values in ansible 2.9

### ISSUE THREAD

**Issue by pmorris3** [ansible-collections/ansible-consul#338]
After an upgrade from 2.8.6 to ansible 2.9, the retry_join value in config.json is empty.  This value is properly being populated in ansible 2.8.6. This may affect further values in config.json as it seems to have something to do with changed template behavior.  I can not yet confirm this either way.

This was experimentation was done off master, as of commit: cc4b842072b3d5095f8c110c9865ed35c481aa39

**Steps to replicate**:
You should be able to replicate this with a reasonably standard configuration, by just upgrading ansible to 2.9

**Observations**:
Pre-Ansible update, on 2.8.6. the output looked like this:
```
    "retry_interval_wan": "30s",
    "retry_join": [
        "192.168.1.10",
        "192.168.1.20",
        "192.168.1.30"
    ],
    "retry_max": 0,
```
I've replaced the IP's above with generic numbers.

After the update, it looks like this:
```
    "retry_interval_wan": "30s",
    "retry_join": [],
    "retry_max": 0,
```

With debug, I was able to express the values for consul_servers, _consul_lan_servers, just before the configuration templating happens.  Inside the copy block, the lookup returns these values as empt lists, if they are directly added to the template file.  Given their use in how retry_join is defined, I am guessing this is the reason.  I do not yet know what change in ansible 2.9 caused this issue.

**Comment by jpiron** [comment]
This is related to https://github.com/ansible/ansible/issues/63940.
Hopefully the fix will be released in next Ansible version .

**Comment by pmorris3** [comment]
It does appear that ansible 2.9.1 has resolved this issue, so I am guessing your reasoning was correct.  I my config with 2.9.1 and without your workaround in place it seems to have worked as expected.

**Comment by brianshumate** [comment]
Thanks for sorting this out here for the record!

Can we close this since the issue is with Ansible v2.9.0 itself and not the role / can be resolved by upgrading to Ansible v2.9.1?

**Comment by pmorris3** [comment]
You're welcome!  There are workarounds, and an answer in the form up upgrading Ansible.  I can't see why this would need to stay open.
